e2e/quit-on-close.e2e.spec.ts

```diff
@@ -0,0 +1,59 @@
+import path from 'path';
+
+import { test, expect, _electron, ElectronApplication } from '@playwright/test';
+
+import { createDefaultSettings, packageLogs, startRancherDesktop } from './utils/TestUtils';
+
+// const sleep = util.promisify(setTimeout);
+
+/**
+ * Using test.describe.serial make the test execute step by step, as described on each `test()` order
+ * Playwright executes test in parallel by default and it will not work for our app backend loading process.
+ * */
+test.describe.serial('quitOnClose setting', () => {
+  test.afterAll(async() => {
+    await packageLogs(__filename);
+  });
+
+  test('should quit when quitOnClose is true and window is closed', async() => {
+    createDefaultSettings({ window: { quitOnClose: true } });
+    const { electronApp } = await startRancherDesktop(__filename, false);
+
+    await expect(closeWindowsAndCheckQuit(electronApp)).resolves.toBe(true);
+  });
+
+  test('should not quit when quitOnClose is false and window is closed', async() => {
+    createDefaultSettings({ window: { quitOnClose: false } });
+    const { electronApp } = await startRancherDesktop(__filename, true);
+
+    await expect(closeWindowsAndCheckQuit(electronApp)).resolves.toBe(false);
+    const tracePath = path.join(__dirname, 'reports', `${ path.basename(__filename) }-quitOnCloseFalse.zip`);
+
+    electronApp.context().tracing.stop({ path: tracePath });
+    await electronApp.close();
+  });
+});
+
+// Closes all of the windows in a running app. Returns a promise that
+// resolves to true when the app has quit within a certain period of time,
+// or that resolves to false when the app does not quit within that period
+// of time.
```

Comment:
Please use `/**` instead of `//` for function documentation (to mark it as JSDoc / JavaDoc style).